### ISSUE TITLE
Implement saving DPI information for HiDPI screenshots

### ISSUE THREAD

**Issue by Flimm** [shutter-project/shutter#577]
Some screens have such a high resolution, that on Linux it is customary to scale the display. This is called [HiDPI](https://wiki.archlinux.org/title/HiDPI) or High-DPI (or Retina on Apple devices).

Here's an example display configuration:

![image](https://user-images.githubusercontent.com/355454/230633609-9dc70eff-97ea-4670-807d-1e1bb306c064.png)

This screen can display 2560x1440 pixels. When the scale is set to 200%, then the display is set to use 1280x720 virtual pixels, while in actuality are 2560x1440 real pixels. In the browser, `screen.width` returns 1280 and `screen.height` returns 720, even though 2560x1440 real pixels are used.

When I take a screenshot in Shutter, then Shutter saves an image that is 2560x1440 pixels big. As far as I can tell, it doesn't save metadata to indicate that this is a HiDPI screenshot that is meant to be displayed as 1280x720 virtual pixels. Now when I use it in other places, the screenshot appears double as a big as I intended. For example, if I attach the screenshot to a document or email or Stack Overflow post, the screenshot appears double as big as I expected it to be displayed.

It would be handy if Shutter had an easy way to scale the image down by 50% when I'm on a HiDPI screen, but as far as I can tell, Shutter does not have this feature.

Some screenshot programs save metadata in the image to indicate that the image should be scaled properly when displayed. In particular, the "X Resolution" and "Y Resolution" EXIF metadata can be set to indicate DPI. Some screenshot programs name the file with a `@2x` suffix, which some image viewers take into consideration when viewing the screenshot. Reference: [Fixing High-DPI Retina screenshot metadata](https://mybyways.com/blog/fixing-high-dpi-retina-screenshot-metadata). I think it would be useful for Shutter to save this metadata, even if not many programs on Linux seem to take note of it, yet.

(I'm using Shutter 0.99.2 on Ubuntu 22.10 with Wayland)

**Comment by Photon89** [comment]
I fear, this is currently not supported. What you could do is activate the automatic thumbnail creation with "50" as size which then will get you both the full size and the scaled down to 50% image. Then you would just have to delete the one that you don't need. You can also create two profiles, one where this option is activated and one where it is not, and then switch the profile from the status bar.

**Comment by Flimm** [comment]
> What you could do is activate the automatic thumbnail creation with "50" as size which then will get you both the full size and the scaled down to 50% image.

I see what you mean. You're talking about enabling the "Thumnbail" option in Shutter's preferences under the "Actions" tab and setting it to 50. This will create a thumbnail image in the target directory for each newly created screenshot that is 50% as wide and 50% as high. Both the full size image and the thumbnail get added to Shutter's tabs, so you can choose which one of the two you want to edit.

That's a handy workaround. Thank you!

**Comment by Photon89** [comment]
Yes, exactly, that's what I meant! :smiley:  I think, this question can be transformed into a feature request (just for completeness, don't think that it will get implemented any time soon...).
